plugins/spotify/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/spotify/MusicResource.java

```diff
@@ -168,103 +171,189 @@ public String setDevice(@PathParam("deviceValue") String deviceValue) {
+	 */
+	@POST
+	@Path("search/album/{searchText}")
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+	public AlbumEntity[] searchAlbums(@PathParam("searchText") String searchText,
+			@QueryParam("limit") @DefaultValue("5") int limit) {
+		List<AlbumEntity> albumList = this.search.searchforAlbums(searchText, limit);
+		if (!albumList.isEmpty()) {
+			return albumList.toArray(new AlbumEntity[albumList.size()]);
 		}
-		throw new WebApplicationException("No results found", Status.NO_CONTENT);
+		throw new WebApplicationException(NO_RESULTS_FOUND, Status.NO_CONTENT);
 	}
 
 	/**
-	 * plays the given music
+	 * this method allows to search for artists
 	 * 
-	 * @param songNumber
-	 *            which song or playlist number should be played
-	 * @param type
-	 *            where to play: allowed parameters: user (playlist), featured (playlist), track
+	 * @param searchText
+	 *            the text you want to search
 	 * @param limit
-	 *            limit of returned tracks or playlists
-	 * @param music
-	 *            the music to be played
-	 * @return the playing music if there is one
+	 *            how many results maximal searched for
+	 * @return a array with ArtistEntities
 	 */
 	@POST
-	@Path("play")
-	@Consumes(MediaType.APPLICATION_JSON)
-	@Produces(MediaType.TEXT_PLAIN)
-	public String play(MusicEntity music, @QueryParam("songNumber") @DefaultValue("0") int songNumber,
-			@QueryParam("type") @DefaultValue("") String type, @QueryParam("limit") @DefaultValue("5") int limit) {
+	@Path("search/artist/{searchText}")
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+	public ArtistEntity[] searchArtists(@PathParam("searchText") String searchText,
+			@QueryParam("limit") @DefaultValue("5") int limit) {
+		List<ArtistEntity> albumList = this.search.searchforArtists(searchText, limit);
+		if (!albumList.isEmpty()) {
+			return albumList.toArray(new ArtistEntity[albumList.size()]);
+		}
+		throw new WebApplicationException(NO_RESULTS_FOUND, Status.NO_CONTENT);
+	}
+
+	/**
+	 * plays the playlist with the index from the search
+	 * 
+	 * @param playlistNumber
+	 *           playlist number should be played
+	 * @param type
+	 *            where to play: allowed parameters: user (playlist), featured (playlist), search
+	 * @return a PlaylistEntity
+	 */
+	@POST
+	@Path("play/playlist")
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+	public PlaylistEntity playPlaylist(@QueryParam("index") @DefaultValue("0") int playlistNumber,
+			@QueryParam("type") @DefaultValue("") String type) {
 		switch (type.toLowerCase()) {
 		case "user":
-			Map<String, String> userPlaylist = this.logic.play(songNumber, SearchTypes.USER_PLAYLISTS);
-			if (userPlaylist.isEmpty()) {
-				throw new WebApplicationException("There is no user playlist available.", Status.CONFLICT);
+			PlaylistEntity userPlaylist = this.logic.playPlaylist(playlistNumber, SearchTypes.USER_PLAYLISTS);
```

Comment:
Please only set the SearchType to a local variable in the switch-case block, then do the other stuff afterwards, once. This way you don't have to duplicate the code three times.